app/v2/controllers/validators/RetrieveHistoricFhlUkPropertyAnnualSubmissionValidatorFactory.scala

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright 2023 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package v2.controllers.validators
+
+import api.controllers.validators.Validator
+import api.controllers.validators.resolvers.{ResolveHistoricTaxYear, ResolveNino}
+import api.models.errors.MtdError
+import cats.data.Validated
+import cats.implicits._
+import config.AppConfig
+import v2.models.request.retrieveHistoricFhlUkPropertyAnnualSubmission.RetrieveHistoricFhlUkPropertyAnnualSubmissionRequestData
+
+import javax.inject.{Inject, Singleton}
+
+@Singleton
+class RetrieveHistoricFhlUkPropertyAnnualSubmissionValidatorFactory @Inject() (appConfig: AppConfig) {
+
+  def validator(nino: String, taxYear: String): Validator[RetrieveHistoricFhlUkPropertyAnnualSubmissionRequestData] =
+    new Validator[RetrieveHistoricFhlUkPropertyAnnualSubmissionRequestData] {
+
+      def validate: Validated[Seq[MtdError], RetrieveHistoricFhlUkPropertyAnnualSubmissionRequestData] =
+        (
+          ResolveNino(nino),
+          ResolveHistoricTaxYear(appConfig.minimumTaxHistoric, appConfig.maximumTaxHistoric, taxYear, None, None)
```

Comment:
Here it should be "appConfig.minimumTaxHistoric + 1"

Reply:
based on how the conditions of the ResolveHistoricTaxYear is set up we don't need to have the `+ 1`

Reply:
Then why has it been added in CreateAmendHistoricFhlUkPropertyAnnualSubmissionValidatorFactory?